### PR TITLE
Configurable doublets and vertex clusterizer

### PR DESCRIPTION
**Pull request by felicepantaleo**

### CODE REVIEW

**Comment by felicepantaleo** [comment]
:construction: Validation running at vinavx2.cern.ch:/data/user/fpantale/patatrack-validation/run.E7kpxtNrw7 ...

**Comment by felicepantaleo** [comment]
:construction: Validation running at vinavx2.cern.ch:/data/user/fpantale/patatrack-validation/run.KhPQjKcUNa ...

**Comment by fwyzard** [comment]
Please find a version with conflicts fixed at [cms-patatrack/pull-342](https://github.com/cms-patatrack/cmssw/tree/pull-342) .

File: CalibTracker/SiPixelESProducers/interface/SiPixelGainCalibrationForHLTGPU.h [hunk 1]
```diff
@@ -17,8 +17,6 @@ class SiPixelGainCalibrationForHLTGPU {
   ~SiPixelGainCalibrationForHLTGPU();
 
   const SiPixelGainForHLTonGPU *getGPUProductAsync(cuda::stream_t<>& cudaStream) const;
-  const SiPixelGainForHLTonGPU *getCPUProduct() const { return gainForHLTonHost_;}
-  const SiPixelGainCalibrationForHLT *getOriginalProduct() { return gains_;}
```
Comment (fwyzard) [inline CalibTracker/SiPixelESProducers/interface/SiPixelGainCalibrationForHLTGPU.h]: In fact, are you suggesting to drop all of 0f57c788619c15c40083e76d05968608bd0d8561 ?

**Comment by fwyzard** [comment]
From the validation, I see some non-negligible (?) changes to the .52 workflows:
*RelValTTbar_13*

  | reference-10824.5 | development-10824.5 | development-10824.52 | testing-10824.52
-- | -- | -- | -- | --
Efficiency | 0.4818 | 0.4917 | *0.5869* | *0.5717*
Number of TrackingParticles (after cuts) | 5556 | 5333 | 5333 | 5333
Number of matched TrackingParticles | 2677 | 2622 | 3130 | 3049
Fake rate | 0.0519 | 0.0517 | 0.0344 | 0.0237
Duplicate rate | 0.0168 | 0.0175 | 0.0002 | 0.0002
Number of tracks | 32452 | 32480 | 43906 | 40590
Number of true tracks | 30769 | 30801 | 42394 | 39626
Number of fake tracks | 1683 | 1679 | 1512 | 964
Number of pileup tracks | 27093 | 27118 | 37688 | 35107
Number of duplicate tracks | 546 | 567 | 10 | 10

*RelValZMM_13*

  | reference-10824.5 | development-10824.5 | development-10824.52 | testing-10824.52
-- | -- | -- | -- | --
Efficiency | 0.5594 | 0.5685 | *0.6422* | *0.6383*
Number of TrackingParticles (after cuts) | 3899 | 3801 | 3801 | 3801
Number of matched TrackingParticles | 2181 | 2161 | 2441 | 2426
Fake rate | 0.0076 | 0.0073 | 0.0065 | 0.0060
Duplicate rate | 0.0136 | 0.0114 | 0.0000 | 0.0000
Number of tracks | 3679 | 3682 | 4593 | 4516
Number of true tracks | 3651 | 3655 | 4563 | 4489
Number of fake tracks | 28 | 27 | 30 | 27
Number of pileup tracks | 0 | 0 | 0 | 0
Number of duplicate tracks | 50 | 42 | 0 | 0


What is "development" here ?

**Comment by fwyzard** [comment]
replaced by #347 .

**Pull request closed without merging**